metal/cloud.go

```diff
@@ -42,6 +42,7 @@ type cloudService interface {
 
 type cloudInstances interface {
 	cloudprovider.Instances
```

Comment:
I think this might be needed for existing installs on older k8s versions to be able to update to this version of CPEM.

Reply:
> I think this might be needed for existing installs on older k8s versions to be able to update to this version of CPEM.

I don't think that is the case, as far as I can tell the use (or not) of Instances/Instancesv2 is isolated to the controller manager code that we are importing from k8s.io/cloud-provider, initializing and running from within the binary that we ship. All use of the Instances/Instancesv2 interfaces appears to done at runtime and the instantiated types don't appear to be persisted anywhere.

Since the instantiated types are not used by external code or persisted anywhere and I'm not seeing any options to override the behavior and force the use of Instances over Instancesv2 when Instancesv2 is implemented, it appears that keeping around support for Instances would just leave around vestigial and unused code.

Reply:
Agreed. @displague do you mind ripping it out? I can run tests against slightly older and slightly newer clusters, if need be.

Reply:
closing this thread since Instances v1 has been ripped out.  There may be some dead code paths that I didn't catch, and some of the node metadata lookup tests may be redundant (as a result of old tests being refactored to fit the new patterns). I think we can clean that up in the future.